src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -677,6 +677,30 @@ func (s *commitLogSource) bootstrapShardSnapshots(
 	blockSize time.Duration,
 	mostRecentCompleteSnapshotByBlockShard map[xtime.UnixNano]map[uint32]fs.FileSetFile,
 ) error {
+	// NB(bodu): We use info files on disk to check if a snapshot should be loaded in as cold or warm.
+	// We do this instead of cross refing blockstarts and current time to handle the case of bootsrapping a
+	// once warm block start after a node has been shut down for a long time. We consider all block starts we
+	// haven't flushed data for yet a warm block start.
+	fsOpts := s.opts.CommitLogOptions().FilesystemOptions()
+	readInfoFilesResults := fs.ReadInfoFiles(fsOpts.FilePathPrefix(), ns.ID(), shard,
+		fsOpts.InfoReaderBufferSize(), fsOpts.DecodingOptions())
+	shardBlockStartsOnDisk := make(map[time.Time]struct{})
```

Comment:
We avoid using `map[time.Time]...` all over the code base because sometimes the same time resolves to a different key based on the Timezone that's kept in `time.Time` so can we make this a `map[xtime.UnixNanos]...` like we use in other places?